rdeditor/rdEditor.py

```diff
@@ -209,242 +215,387 @@ def closeEvent(self, event):
         event.ignore()
 
     def exitFile(self):
-        response = self.msgApp("Confirmation","This will quit the application. Do you want to Continue?")
+        response = self.msgApp(
+            "Confirmation", "This will quit the application. Do you want to Continue?"
+        )
         if response == "Y":
             self.ptable.close()
-            exit(0) #TODO, how to exit qapplication from within class instance?
+            exit(0)  # TODO, how to exit qapplication from within class instance?
         else:
             self.editor.logger.debug("Abort closing")
 
-
     # Function to show Diaglog box with provided Title and Message
-    def msgApp(self,title,msg):
-        userInfo = QMessageBox.question(self,title,msg,
-                                        QMessageBox.Yes | QMessageBox.No)
+    def msgApp(self, title, msg):
+        userInfo = QMessageBox.question(
+            self, title, msg, QMessageBox.Yes | QMessageBox.No
+        )
         if userInfo == QMessageBox.Yes:
             return "Y"
         if userInfo == QMessageBox.No:
             return "N"
         self.close()
 
     def aboutHelp(self):
-        QMessageBox.about(self, "About Simple Molecule Editor",
-                """A Simple Molecule Editor where you can edit molecules\nBased on RDKit! http://www.rdkit.org/ \nSome icons from http://icons8.com\n\nSource code: https://github.com/EBjerrum/rdeditor""")
+        QMessageBox.about(
+            self,
+            "About Simple Molecule Editor",
+            """A Simple Molecule Editor where you can edit molecules\nBased on RDKit! http://www.rdkit.org/ \nSome icons from http://icons8.com\n\nSource code: https://github.com/EBjerrum/rdeditor""",
+        )
 
     def setAction(self):
         sender = self.sender()
         self.editor.setAction(sender.objectName())
-        self.myStatusBar.showMessage("Action %s selected"%sender.objectName())
+        self.myStatusBar.showMessage("Action %s selected" % sender.objectName())
+
+    def setRingType(self, ringType):
+        if ringType == self.editor.ringtype:
+            self.editor.setRingType(None)
+            self.bezeneAction.setChecked(False)
+            self.ringSixAction.setChecked(False)
```

Comment:
This is not the way. The actions should be added to an action group, so that only one can be selected at a time. It should not be necessary to do manually